dev/Controller/MockController.php

```diff
@@ -0,0 +1,152 @@
+<?php
+/**
+ * Copyright 2019 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Dev\Controller;
+
+use Dev\Mock\MockGateway;
+use Exception;
+use SAML2\Constants;
+use SAML2\Response as SamlResponse;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Twig\Environment;
+
+class MockController extends AbstractController
+{
+    /**
+     * @var MockGateway
+     */
+    private $mockStepupGateway;
+    /**
+     * @var Environment
+     */
+    private $twig;
+
+    public function __construct(MockGateway $mockStepupGateway, Environment $twig)
+    {
+        $this->mockStepupGateway = $mockStepupGateway;
+        $this->twig = $twig;
+    }
+
+    /**
+     * This is the sso action used to mock a GSSP callout
+     *
+     * @Route(
+     *     "/mock/sso",
+     *     name="mock_sso",
+     *     methods={"GET"}
+     * )
+     *
+     * @param Request $request
+     * @return string|Response
+     */
+    public function ssoAction(Request $request)
+    {
+        if (!in_array($this->getParameter('kernel.environment'), ['test', 'dev'])) {
+            throw new Exception('Invalid environment encountered.');
+        }
+
+        try {
+            // Check binding
+            if (!$request->isMethod(Request::METHOD_GET)) {
+                throw new BadRequestHttpException(sprintf(
+                    'Could not receive AuthnRequest from HTTP Request: expected a GET method, got %s',
+                    $request->getMethod()
+                ));
```

Comment:
This code block should be unreachable now that only `{GET}` methods are allowed?

Reply:
Removed the annotation because the exception is more informative.